src/DistributedLock.Redis/Primitives/RedisSemaphorePrimitive.cs

```diff
@@ -78,4 +78,30 @@ public Task ReleaseAsync(IDatabaseAsync database, bool fireAndForget) =>
     public Task<bool> TryExtendAsync(IDatabaseAsync database) => ExtendScript.ExecuteAsync(database, this).AsBooleanTask();
 
     public bool IsConnected(IDatabase database) => database.IsConnected(this._key, CommandFlags.DemandMaster);
+
+    /// <summary>
+    /// Get Current Count
+    /// </summary>
+    /// <param name="database"></param>
+    /// <returns></returns>
+    internal async ValueTask<int> GetCurrentCountAsync(IDatabaseAsync database)
```

Comment:
Let's call this `GetAcquiredCountAsync` since it doesn't return the remaining count.